src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -659,19 +659,22 @@ private String searchRubyHome(Options options) throws IOException {
             if (isRubyHome(candidate)) {
                 return candidate.getCanonicalPath();
             } else {
-                warning.append("* Path '").
+                warning.append("* path '").
                         append(candidate).
                         append("' derived from GraalVM home '").
                         append(graalVMHome).
-                        append("' does not appear to be TruffleRuby's home.\n");
+                        append("' does not look like TruffleRuby's home\n");
 
             }
         } else {
             warning.append("* GraalVM home not found.\n");
         }
 
-        warning.append("* Try to set home using -Xhome=PATH option.");
-        Log.LOGGER.warning(warning.toString());
+        warning.append("* try to set home using -Xhome=PATH option");
+
+        Log.LOGGER.warning("could not determine TruffleRuby's home - the standard library will not be available");
+
+        Log.LOGGER.config(warning.toString());
```

Comment:
This will make it much harder to debug in CI. We got the case recently in GraalVM where `graalvm.home` was set to an incorrect value and would be quite troublesome to diagnose if the details are not printed.
Since we have this embedded flag, can we not just use that to ignore the home unset warning, but in other cases print the full details since it's quite wrong to not have a home in those scenarios?

Reply:
That's certainly a good idea for the underlying point of this PR - to tidy up embedded situations in the short term...

Reply:
https://github.com/oracle/truffleruby/pull/1134/files#diff-005791c9e6171ae6323f76c2722d331bR268 might be useful for this.

Reply:
Concretely, how about printing the details as warning if `!embedded` and as config otherwise ?

Reply:
Ok.